volatility3/framework/plugins/windows/skeleton_key_check.py

```diff
@@ -0,0 +1,578 @@
+                                  config, 
+                                  config_path: str, 
+                                  proc_layer_name: str, 
+                                  cryptdll_base: int):
+        """
+        Builds a symbol table from the cryptdll types generated after binary analysis
+
+        Args:
+            context: the context to operate upon
+            config: 
+            config_path:
+            proc_layer_name: name of the lsass.exe process layer
+            cryptdll_base: base address of cryptdll.dll inside of lsass.exe
+        """
+        table_mapping = {"nt_symbols":  config["nt_symbols"]}
+
+        cryptdll_symbol_table = intermed.IntermediateSymbolTable.create(context = context, 
+                                                                        config_path = config_path,
+                                                                        sub_path = "windows",
+                                                                        filename = "kerb_ecrypt",
+                                                                        table_mapping = table_mapping)
+
+        return context.module(cryptdll_symbol_table, proc_layer_name, offset = cryptdll_base)
+
+    def _find_and_parse_cryptdll(self, proc_list: Iterable) -> \
+                                Tuple[interfaces.context.ContextInterface, str, int, int]:  
+        """
+        Finds the base address of cryptdll.dll insode of lsass.exe
+
+        Args:
+            proc_list: the process list filtered to just lsass.exe instances
+
+        Returns:
+            A tuple of:
+            lsass_proc: the process object for lsass.exe
+            proc_layer_name: the name of the lsass.exe process layer
+            cryptdll_base: the base address of cryptdll.dll
+            crytpdll_size: the size of the VAD for cryptdll.dll
+        """
+        lsass_proc    = None
+        proc_layer_name = None
+        cryptdll_base = None
+        cryptdll_size  = None
+
+        for proc in proc_list:
+            try:
+                proc_id = proc.UniqueProcessId
+                proc_layer_name = proc.add_process_layer()
+            except exceptions.InvalidAddressException as excp:
+                vollog.debug("Process {}: invalid address {} in layer {}".format(proc_id, excp.invalid_address,
+                                                                                 excp.layer_name))
+                continue
+
+            proc_layer = self.context.layers[proc_layer_name]
+
+            for vad in proc.get_vad_root().traverse():
+                filename = vad.get_file_name()
+                if type(filename) == renderers.NotApplicableValue or not filename.lower().endswith("cryptdll.dll"):
+                    continue
+       
+                cryptdll_base = vad.get_start()
+                cryptdll_size = vad.get_end() - cryptdll_base
+
+                break
+
+            lsass_proc = proc
+            break
```

Comment:
lsass_proc is set to show that we for sure found an lsass.exe processes. Python does let loop variables bleed through after the loop, but I never liked doing that. 

Also, I want to break after the first lsass.exe as windows keeps the process list ordered by when the processes started (new ones get added to the end of list), so in the case of fake lsass.exe, they would be after the real one in the list, unless you had something like EFI/MBR type malware, which is a whole different issue.

Reply:
It was more the multiple breaks and the flow.  I'd have set `lsass_proc` when we found the right value and rather than `break` from one loop and `break` from the second loop, just do a return as soon as we've found what we're looking for?  It might already be different due to the comment above, but it just felt a little inefficient to be breaking out of loops to get a return statement.  No real problem with it if you want to leave it in there...